File: 5 semestre/Programacao para Dispositivos Moveis/UAS/UA1/media_notas.dart
import 'dart:html';

void main() {
    DivElement container = DivElement();
    container.style.width = '400px';
    container.style.margin = '50px auto';
    container.style.padding = '20px';
    container.style.border = '1px solid #ccc';
    container.style.borderRadius = '10px';
    container.style.fontFamily = 'Arial';

    HeadingElement titulo = HeadingElement.h2();
    titulo.text = 'Calculadora de Média';
    titulo.style.textAlign = 'center';

    LabelElement labelN1 = LabelElement();
    labelN1.text = 'Nota 1:';

    InputElement inputN1 = InputElement();
    inputN1.type = 'number';
    inputN1.placeholder = 'Digite a primeira nota';
    inputN1.style.width = '100%';
    inputN1.style.marginBottom = '10px';

    LabelElement labelN2 = LabelElement();
    labelN2.text = 'Nota 2:';

    InputElement inputN2 = InputElement();
    inputN2.type = 'number';
    inputN2.placeholder = 'Digite a segunda nota';
    inputN2.style.width = '100%';
    inputN2.style.marginBottom = '10px';

    ButtonElement botaoCalcular = ButtonElement();
    botaoCalcular.text = 'Calcular Média';
    botaoCalcular.style.width = '100%';
    botaoCalcular.style.padding = '10px';
    botaoCalcular.style.marginTop = '10px';
    botaoCalcular.style.cursor = 'pointer';

    ButtonElement botaoLimpar = ButtonElement();
    botaoLimpar.text = 'Limpar';
    botaoLimpar.style.width = '100%';
    botaoLimpar.style.padding = '10px';
    botaoLimpar.style.marginTop = '10px';
    botaoLimpar.style.cursor = 'pointer';

    ParagraphElement resultado = ParagraphElement();
    resultado.text = 'Resultado aparecerá aqui';
    resultado.style.marginTop = '20px';
    resultado.style.fontWeight = 'bold';
    resultado.style.textAlign = 'center';

    ParagraphElement status = ParagraphElement();
    status.text = '';
    status.style.marginTop = '10px';
    status.style.textAlign = 'center';

    void calcularMedia(Event e) {
        double? n1 = double.tryParse(inputN1.value ?? '');
        double? n2 = double.tryParse(inputN2.value ?? '');

        if (n1 == null || n2 == null) {
        resultado.text = 'Por favor, insira valores válidos!';
        resultado.style.color = 'red';
        status.text = '';
        return;
        }

        double media = (n1 + n2) / 2;

        resultado.text = 'Média: ${media.toStringAsFixed(2)}';
        resultado.style.color = 'green';

        if (media >= 7) {
        status.text = 'Aprovado';
        status.style.color = 'green';
        } else if (media >= 5) {
        status.text = 'Recuperação';
        status.style.color = 'orange';
        } else {
        status.text = 'Reprovado';
        status.style.color = 'red';
        }
    }

    void limparCampos(Event e) {
        inputN1.value = '';
        inputN2.value = '';
        resultado.text = 'Resultado aparecerá aqui';
        resultado.style.color = 'black';
        status.text = '';
    }

    inputN1.onInput.listen((event) {
        inputN1.style.border = '1px solid #ccc';
    });

    inputN2.onInput.listen((event) {
        inputN2.style.border = '1px solid #ccc';
    });

    botaoCalcular.onClick.listen(calcularMedia);
    botaoLimpar.onClick.listen(limparCampos);

    container.children.add(titulo);
    container.children.add(labelN1);
    container.children.add(inputN1);
    container.children.add(labelN2);
    container.children.add(inputN2);
    container.children.add(botaoCalcular);
    container.children.add(botaoLimpar);
    container.children.add(resultado);
    container.children.add(status);

    document.body?.children.add(container);

    DivElement rodape = DivElement();
    rodape.text = 'Aplicação Dart Web usando dart:html';
    rodape.style.textAlign = 'center';
    rodape.style.marginTop = '30px';
    rodape.style.fontSize = '12px';
    rodape.style.color = '#777';

    document.body?.children.add(rodape);

    DivElement info = DivElement();
    info.style.marginTop = '20px';
    info.style.fontSize = '14px';

    ParagraphElement p1 = ParagraphElement();
    p1.text = 'Digite duas notas para calcular a média.';

    ParagraphElement p2 = ParagraphElement();
    p2.text = 'A média será exibida junto com o status do aluno.';

    ParagraphElement p3 = ParagraphElement();
    p3.text = 'Este exemplo utiliza eventos e manipulação do DOM com Dart.';

    info.children.add(p1);
    info.children.add(p2);
    info.children.add(p3);

    container.children.add(info);

    DivElement decoracao = DivElement();
    decoracao.style.height = '5px';
    decoracao.style.marginTop = '20px';
    decoracao.style.backgroundColor = '#4CAF50';

    container.children.add(decoracao);
}

File: 5 semestre/Programacao para Dispositivos Moveis/UAS/UA1/java/src/MediaNotas.java
import javax.swing.*;
import java.awt.*;
import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;

public class MediaNotas extends JFrame {

    private JTextField inputN1;
    private JTextField inputN2;
    private JLabel resultado;
    private JLabel status;

    public MediaNotas() {
        setTitle("Calculadora de Média");
        setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
        setSize(450, 500);
        setLocationRelativeTo(null);
        setResizable(false);

        // Painel principal com margens
        JPanel mainPanel = new JPanel();
        mainPanel.setLayout(new BoxLayout(mainPanel, BoxLayout.Y_AXIS));
        mainPanel.setBorder(BorderFactory.createEmptyBorder(20, 20, 20, 20));
        mainPanel.setBackground(Color.WHITE);

        // Título
        JLabel titulo = new JLabel("Calculadora de Média");
        titulo.setFont(new Font("Arial", Font.BOLD, 18));
        titulo.setAlignmentX(Component.CENTER_ALIGNMENT);

        // Nota 1
        JLabel labelN1 = new JLabel("Nota 1:");
        labelN1.setAlignmentX(Component.LEFT_ALIGNMENT);
        labelN1.setFont(new Font("Arial", Font.PLAIN, 14));

        inputN1 = new JTextField();
        inputN1.setMaximumSize(new Dimension(Integer.MAX_VALUE, 30));
        inputN1.setFont(new Font("Arial", Font.PLAIN, 14));
        inputN1.setAlignmentX(Component.LEFT_ALIGNMENT);

        // Nota 2
        JLabel labelN2 = new JLabel("Nota 2:");
        labelN2.setAlignmentX(Component.LEFT_ALIGNMENT);
        labelN2.setFont(new Font("Arial", Font.PLAIN, 14));

        inputN2 = new JTextField();
        inputN2.setMaximumSize(new Dimension(Integer.MAX_VALUE, 30));
        inputN2.setFont(new Font("Arial", Font.PLAIN, 14));
        inputN2.setAlignmentX(Component.LEFT_ALIGNMENT);

        // Botão Calcular
        JButton botaoCalcular = new JButton("Calcular Média");
        botaoCalcular.setAlignmentX(Component.CENTER_ALIGNMENT);
        botaoCalcular.setMaximumSize(new Dimension(Integer.MAX_VALUE, 40));
        botaoCalcular.setFont(new Font("Arial", Font.BOLD, 14));
        botaoCalcular.setBackground(new Color(76, 175, 80));
        botaoCalcular.setForeground(Color.WHITE);
        botaoCalcular.setFocusPainted(false);
        botaoCalcular.setCursor(Cursor.getPredefinedCursor(Cursor.HAND_CURSOR));

        // Botão Limpar
        JButton botaoLimpar = new JButton("Limpar");
        botaoLimpar.setAlignmentX(Component.CENTER_ALIGNMENT);
        botaoLimpar.setMaximumSize(new Dimension(Integer.MAX_VALUE, 40));
        botaoLimpar.setFont(new Font("Arial", Font.BOLD, 14));
        botaoLimpar.setBackground(new Color(200, 200, 200));
        botaoLimpar.setForeground(Color.BLACK);
        botaoLimpar.setFocusPainted(false);
        botaoLimpar.setCursor(Cursor.getPredefinedCursor(Cursor.HAND_CURSOR));

        // Label resultado
        resultado = new JLabel("Resultado aparecerá aqui");
        resultado.setAlignmentX(Component.CENTER_ALIGNMENT);
        resultado.setFont(new Font("Arial", Font.BOLD, 14));

        // Label status
        status = new JLabel(" ");
        status.setAlignmentX(Component.CENTER_ALIGNMENT);
        status.setFont(new Font("Arial", Font.BOLD, 14));

        // Informações adicionais
        JTextArea info = new JTextArea(
            "Digite duas notas para calcular a média.\n" +
            "A média será exibida junto com o status do aluno.\n" +
            "Este exemplo utiliza eventos e manipulação do DOM com Dart."
        );
        info.setEditable(false);
        info.setBackground(null);
        info.setFont(new Font("Arial", Font.PLAIN, 12));
        info.setForeground(Color.DARK_GRAY);
        info.setAlignmentX(Component.LEFT_ALIGNMENT);

        // Rodapé decorativo
        JPanel decoracao = new JPanel();
        decoracao.setBackground(new Color(76, 175, 80));
        decoracao.setMaximumSize(new Dimension(Integer.MAX_VALUE, 5));
        decoracao.setAlignmentX(Component.CENTER_ALIGNMENT);

        // Adiciona componentes ao painel principal
        mainPanel.add(titulo);
        mainPanel.add(Box.createRigidArea(new Dimension(0, 15)));
        mainPanel.add(labelN1);
        mainPanel.add(Box.createRigidArea(new Dimension(0, 5)));
        mainPanel.add(inputN1);
        mainPanel.add(Box.createRigidArea(new Dimension(0, 15)));
        mainPanel.add(labelN2);
        mainPanel.add(Box.createRigidArea(new Dimension(0, 5)));
        mainPanel.add(inputN2);
        mainPanel.add(Box.createRigidArea(new Dimension(0, 20)));
        mainPanel.add(botaoCalcular);
        mainPanel.add(Box.createRigidArea(new Dimension(0, 10)));
        mainPanel.add(botaoLimpar);
        mainPanel.add(Box.createRigidArea(new Dimension(0, 20)));
        mainPanel.add(resultado);
        mainPanel.add(Box.createRigidArea(new Dimension(0, 5)));
        mainPanel.add(status);
        mainPanel.add(Box.createRigidArea(new Dimension(0, 20)));
        mainPanel.add(info);
        mainPanel.add(Box.createRigidArea(new Dimension(0, 10)));
        mainPanel.add(decoracao);

        add(mainPanel);

        // Ações dos botões
        botaoCalcular.addActionListener(new ActionListener() {
            @Override
            public void actionPerformed(ActionEvent e) {
                calcularMedia();
            }
        });

        botaoLimpar.addActionListener(new ActionListener() {
            @Override
            public void actionPerformed(ActionEvent e) {
                limparCampos();
            }
        });

        // Restaura borda padrão ao digitar
        inputN1.addCaretListener(e -> inputN1.setBorder(UIManager.getBorder("TextField.border")));
        inputN2.addCaretListener(e -> inputN2.setBorder(UIManager.getBorder("TextField.border")));
    }

    private void calcularMedia() {
        String textoN1 = inputN1.getText().trim();
        String textoN2 = inputN2.getText().trim();

        if (textoN1.isEmpty() || textoN2.isEmpty()) {
            resultado.setText("Por favor, insira valores válidos!");
            resultado.setForeground(Color.RED);
            status.setText("");

            return;
        }

        try {
            double n1 = Double.parseDouble(textoN1);
            double n2 = Double.parseDouble(textoN2);

            double media = (n1 + n2) / 2.0;

            resultado.setText(String.format("Média: %.2f", media));
            resultado.setForeground(new Color(0, 128, 0));

            if (media >= 7.0) {
                status.setText("Aprovado");
                status.setForeground(new Color(0, 128, 0));
            } 
            
            else if (media >= 5.0) {
                status.setText("Recuperação");
                status.setForeground(Color.ORANGE);
            } 
            
            else {
                status.setText("Reprovado");
                status.setForeground(Color.RED);
            }
        } 
        
        catch (NumberFormatException ex) {
            resultado.setText("Por favor, insira valores numéricos válidos!");
            resultado.setForeground(Color.RED);
            status.setText("");
        }
    }

    private void limparCampos() {
        inputN1.setText("");
        inputN2.setText("");
        resultado.setText("Resultado aparecerá aqui");
        resultado.setForeground(Color.BLACK);
        status.setText("");
        inputN1.setBorder(UIManager.getBorder("TextField.border"));
        inputN2.setBorder(UIManager.getBorder("TextField.border"));
    }

    public static void main(String[] args) {
        SwingUtilities.invokeLater(() -> {
            new MediaNotas().setVisible(true);
        });
    }
}
